plugin-catalog/src/components/plugins/Detail.tsx

```diff
@@ -0,0 +1,275 @@
+
+  const fetchStatus = async () => {
+    try {
+      let status = await PluginManager.getStatus(identifier);
+      if (status.type === 'error' && status.message === 'No such operation in progress') {
+        setCurrentActionState(null);
+        setCurrentActionProgress(0);
+        setCurrentAction(null);
+        return;
+      }
+      while (status && currentAction) {
+        if (status.type === 'error' || status.type === 'success') {
+          const enrichedPluginData = await checkIfPluginIsInstalled(pluginDetail);
+          setPluginDetail(enrichedPluginData);
+          setCurrentActionState(null);
+          setCurrentActionMessage(null);
+          setCurrentActionProgress(0);
+          setCurrentAction(null);
+          break;
+        }
+        if (status.percentage !== undefined) {
+          setCurrentActionProgress(status.percentage);
+        }
+        setCurrentActionState(status.type);
+        setCurrentActionMessage(status.message);
+
+        await new Promise(resolve => setTimeout(resolve, 1000));
+        status = await PluginManager.getStatus(identifier);
+      }
+    } catch (error) {
+      setCurrentActionState(null);
+      setCurrentActionMessage(null);
+      setCurrentActionProgress(0);
+      console.error('Error:', error);
+    }
+  };
+
+  useEffect(() => {
+    let isMounted = true;
+
+    const initialize = async () => {
+      if (isMounted) {
+        const data = await fetchHeadlampPluginDetail(repoName, pluginName);
+        const enrichedPluginData = await checkIfPluginIsInstalled(data);
+        setPluginDetail(enrichedPluginData);
+      }
+
+      fetchStatus();
+    };
+
+    initialize();
+
+    return () => {
+      isMounted = false;
+    };
+  }, [repoName, pluginName, currentAction]);
+
+  useEffect(() => {
+    if (currentAction) {
+      fetchStatus(); // Fetch status when an action is triggered
+    }
+  }, [currentAction]);
+
+  const handleInstall = () => {
+    setCurrentAction('Install');
+    PluginManager.install(identifier, url);
+  };
+
+  const handleUpdate = (pluginName: string) => {
+    setCurrentAction('Update');
+    PluginManager.update(identifier, pluginName);
+  };
+
+  const handleUninstall = (pluginName: string) => {
+    setCurrentAction('Uninstall');
+    PluginManager.uninstall(identifier, pluginName);
+  };
+
+  return (
+    <>
+      <SectionBox
+        title={
+          <SectionHeader
+            title={pluginDetail ? pluginDetail.display_name : pluginName}
+            actions={[
+              currentAction === null ? (
+                pluginDetail ? (
+                  pluginDetail.isInstalled ? (
+                    <>
+                      <ActionButton
+                        description="Upgrade"
+                        onClick={() => handleUpdate(pluginDetail.packageName)}
+                        icon="mdi:arrow-up-bold"
+                        iconButtonProps={{ disabled: !pluginDetail.updateAvailable }}
+                      />
```

Comment:
In alignment with what we do in the UI, do not show this button if there's no upgrade available. And let's label it "Update" instead of "Upgrade".